File: homework/hw1-lushenkov/internal/domain/animal/tiger.go
package animal

type Tiger struct{ *Predator }

func NewTiger(name string, foodPerDayKg int) (*Tiger, error) {
	p, err := NewPredator(name, "Tiger", foodPerDayKg)
	if err != nil {
		return nil, err
	}
	return &Tiger{Predator: p}, nil
}

File: homework/src/main/java/hse/kpo/homework1/living/Tiger.java
package hse.kpo.homework1.living;

import java.util.Map;

/**
 * Конкретный класс для тигров (хищное животное).
 * Следует принципу Liskov Substitution Principle (LSP):
 * может использоваться везде, где ожидается Predator или Animal.
 */
public class Tiger extends Predator {
    public Tiger(String name, int food, int number) {
        super(name, food, number);
    }

    public Tiger(Map<String, Object> params) {
        this(
            (String) params.get("name"),
            (Integer) params.get("food"),
            (Integer) params.get("inventoryNumber")
        );
    }
}
